src/control/client/network.go

```diff
@@ -0,0 +1,173 @@
+//
+// (C) Copyright 2018-2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+
+	ctlpb "github.com/daos-stack/daos/src/control/common/proto/ctl"
+)
+
+const (
+	networkScanTimeout = 10 * time.Second
+)
+
+type ProviderListResult struct {
+	providerList string
+}
+
+type NetworkScanResult struct {
+	provider string
+	device   string
+	numanode uint
+	err      error
+}
+
+type NetworkScanResultMap map[string][]NetworkScanResult
+
+func (nsrm NetworkScanResultMap) String() string {
+	var buf bytes.Buffer
+	servers := make([]string, 0, len(nsrm))
+
+	for server := range nsrm {
+		servers = append(servers, server)
+	}
+	sort.Strings(servers)
+
+	for _, server := range servers {
+		fmt.Fprintf(&buf, "%s:\n%s\n", server, nsrm[server])
```

Comment:
does this work as expected, you are passing []NetworkScanResult which I don't think has a Stringer interface implementation, maybe works for a slice by default? otherwise you could just iterate through the slice here and pass each result to the formatter as the results have a stringer interface implementation

Reply:
It does generate the output I expected.